Modules/_datetimemodule.c

```diff
@@ -845,6 +845,17 @@ parse_isoformat_time(const char *dtstr, size_t dtlen, int *hour, int *minute,
     return rv ? -5 : 1;
 }
 
+static void
+isocalendar_to_ymd(int* year, int* month_week, int* day) {
+    // Convert (Y, W, D) to (Y, M, D) in-place
+    int day_1 = iso_week1_monday(*year);
+
+    int day_offset = (*month_week - 1)*7 + *day - 1;
+
+    ord_to_ymd(day_1 + day_offset, year, month_week, day);
```

Comment:
This will overflow (and core dump) if the year is very big. Example:

```python
>>> import datetime
>>> datetime.date.fromisocalendar(9999999,1,1)
python: /home/pablogsal/github/cpython/Modules/_datetimemodule.c:307: ord_to_ymd: Assertion `ordinal >= 1' failed.
[1]    19212 abort (core dumped)  ./python
```

Reply:
Hm, I was under the impression that `i` arguments to the `PyArg_Parse*` macros were checked for overflow, maybe I got that wrong?

The only reason I didn't have a test for this yet was that I hadn't decided whether it was fine to surface an `OverflowError` or whether I should catch and re-raise `ValueError`, since any overflow is a domain error anyway. I think catch and re-raise is the right way to go.

I'll update the PR tomorrow.

Reply:
Hm, I can't reproduce the core dump, but actually I realize what's happening here, it's not that any of the inputs is overflowing, it's that the number of ordinal days that is overflowing. That's easy enough to fix by more aggressively checking the bounds on the input.

Reply:
The inputs themselves cannot overflow, if you try to pass a bigger value of the year you will receive different error messaging telling that a C `int` cannot be constructed. Is the moment you are adding two valid c ints (`day_1 + day_offset`) the moment it overflows.

Reply:
@pablogsal Yeah, I think the general gist of what you are saying is correct, but I think it's actually overflowing when the year gets converted to ordinal date, not the addition (which can add a maximum of ~400 days because of the bounds of the inputs). Either way, it's fixed now.